Lombiq.Analyzers.PowerShell/Rules/Measure-AutomaticVariableAlias/Measure-AutomaticVariableAlias.psm1

```diff
@@ -0,0 +1,69 @@
+﻿<#
+.SYNOPSIS
+    Replaces usages of the alias of the automatic variable ($_) with its original form ($PSItem).
+.DESCRIPTION
+    Replaces usages of the alias of the automatic variable ($_) with its original form ($PSItem), similarly to the
+    AvoidUsingCmdletAliases rule.
+.EXAMPLE
+    Measure-AutomaticVariableAlias -Token $Token
+.INPUTS
+    [System.Management.Automation.Language.Token[]]
+.OUTPUTS
+    [Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]]
+.NOTES
+    Copied (and modified version of)
+    https://github.com/PowerShell/PSScriptAnalyzer/blob/master/Tests/Engine/CommunityAnalyzerRules/CommunityAnalyzerRules.psm1#L613.
+#>
+function Measure-AutomaticVariableAlias
+{
+    [CmdletBinding()]
+    [OutputType([Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]])]
+    Param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.Management.Automation.Language.Token[]]
+        $Token
+    )
+
+    Process
+    {
+        $results = @()
+
+        try
+        {
+            # Filter down tokens to just variable tokens with the name "_".
+            foreach ($automaticVariableAliasToken in $Token | Where-Object { $PSItem.GetType().Name -eq "VariableToken" -and $PSItem.Name -eq "_" })
+            {
+                $correctionTypeName = "Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent"
+                $correctionExtent = New-Object -TypeName $correctionTypeName -ArgumentList @(
+                    $automaticVariableAliasToken.Extent.StartLineNumber
+                    $automaticVariableAliasToken.Extent.EndLineNumber
+                    $automaticVariableAliasToken.Extent.StartColumnNumber
+                    $automaticVariableAliasToken.Extent.EndColumnNumber
+                    "`$PSItem"
+                    "Replaced the usage of the alias of the automatic variable '`$_' with its full content '`$PSItem'."
```

Comment:
I think here it's justified to use single quotes, because you are just manually preventing variable evaluation. Also the inner quotes should be double anyway. Same below for the Message's value.

Reply:
I wanted to do that, but this is how the built-in messages are formatted, e.g.:
`'%' is an alias of 'ForEach-Object'. Alias can introduce possible problems and make scripts hard to maintain. Please consider changing alias to its full content.`

On the other hand, if this whole string in encased by apostrophe, then $ doesn't escaping, but outputting an apostrophe character requires it to be escaped by another one, which I think is still better, so I'll do that.